### PR TITLE
Semantic other checks

### PR DESCRIPTION
**Pull request by N-Maas**
Third round of semantic checks

### CODE REVIEW

File: src/main/java/edu/kit/compiler/semantic/SemanticChecks.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+                boolean returns = method.getStatementBlock().accept(visitor);
+                if (!returns && !method.getType().equals(voidType)) {
+                    errorHandler.receive(new SemanticError(method,
+                        "return statement for all branches required"));
+                }
+            }
+            for (MethodNode method: currentClass.getStaticMethods()) {
+                MethodCheckVisitor visitor = new MethodCheckVisitor(true, errorHandler, stringClass);
+                method.getStatementBlock().accept(visitor);
+            }
```
Comment (g4rgoil) [inline src/main/java/edu/kit/compiler/semantic/SemanticChecks.java:62]: You could consider checking for `return` on all paths in static methods as well.

File: src/main/java/edu/kit/compiler/semantic/SemanticChecks.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    }
+
+    public Boolean visit(ValueExpressionNode expr) {
+        // nothing to check here
+        return false;
+    }
+
+    public Boolean visit(ThisExpressionNode expr) {
+        if (this.isMain) {
+            errorHandler.receive(new SemanticError(expr,
```
Comment (larsk21) [inline src/main/java/edu/kit/compiler/semantic/SemanticChecks.java:201]: Already checked in second phase

File: src/main/java/edu/kit/compiler/semantic/SemanticException.java [hunk 3]
```diff
@@ -32,4 +45,8 @@ public SemanticException(SemanticError error) {
     public JavaEasyCompiler.Result getResult() {
         return JavaEasyCompiler.Result.SemanticError;
     }
+
+    public boolean surpressMessage() {
```
Comment (larsk21) [inline src/main/java/edu/kit/compiler/semantic/SemanticException.java:49]: From the name, I would expect that `supressMessage` sets `quiet` to true.

**Review by larsk21 — APPROVED** [review]
Failing tests are not caused by third phase.

**Pull request merged**